Programowanie3/Assets/Scripts/Weapon.cs

```diff
@@ -25,12 +27,17 @@ public void Shoot()
             switch (shootType)
             {
                 case ShootType.Bullet:
-                    Bullet shotBullet = Instantiate(bulletPrefab, shootPoint.position, shootPoint.rotation);
-                    shotBullet.Launch(bulletsSpeed, range, damage);
+                    for (int i = 0; i < numberOfBullets; i++)
+                    {
+                        
+                        Quaternion spreadRotation = Quaternion.Euler(0, Random.Range(-spreadAngle, spreadAngle), 0) * Quaternion.Euler(Random.Range(-spreadAngle, spreadAngle), 0, 0);                        
```

Comment:
```suggestion
                        Quaternion spreadRotation = Quaternion.Euler(0, Random.Range(-spreadAngle /2 , spreadAngle / 2), 0) );                        
```
Jak mamy grę w perspektywie top-down, to rozrzut w pionie słabo się sprawdza, bo nie możemy celować w pionie i pociski przelatują nad przeciwnikami nie z naszej winy. To co napisałeś lepiej działałoby w FPSie.
Drugą sprawą jest wartość spreadAngle. Teraz broń ma rozrzut spreadAngle *2. Jak masz spreadAngle = 10, to możesz odchyliś od -10 do +10, czyli w sumie o 20. Powinieneś albo dzielić przez 2, albo gdzieś zaznaczyć że to spreadAngle w jedną stronę.